wagtail/tests/utils/page_tests.py

```diff
@@ -0,0 +1,115 @@
+from django.test import TestCase
+from django.urls import reverse
+from django.utils.text import slugify
+
+from .wagtail_tests import WagtailTestUtils
+
+
+class WagtailPageTests(WagtailTestUtils, TestCase):
+    """
+    A set of asserts to help write tests for your own Wagtail site.
+    """
+    def setUp(self):
+        super().setUp()
+        self.login()
+
+    def _testCanCreateAt(self, parent_model, child_model):
+        return child_model in parent_model.allowed_subpage_models()
+
+    def assertCanCreateAt(self, parent_model, child_model, msg=None):
+        """
+        Assert a particular child Page type can be created under a parent
+        Page type. ``parent_model`` and ``child_model`` should be the Page
+        classes being tested.
+        """
+        if not self._testCanCreateAt(parent_model, child_model):
+            msg = self._formatMessage(msg, "Can not create a %s.%s under a %s.%s" % (
+                child_model._meta.app_label, child_model._meta.model_name,
+                parent_model._meta.app_label, parent_model._meta.model_name))
+            raise self.failureException(msg)
```

Comment:
I know this has just been moved, but I’m wondering: why not use this instead?
```python
self.assertTrue(child_model.can_create_at(parent_model),
                msg='Cannot create a %s under %s' % (
                    child_model._meta.label, parent_model._meta.label))
```

Since `Page.can_create_at` can contain some additional custom functionalities.